cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/BatchETLDBAdapterTest.java

```diff
@@ -268,6 +267,81 @@ public void testDBSource() throws Exception {
     Assert.assertEquals("user2", Bytes.toString(row2.get("BLOB_COL"), 0, 5));
     Assert.assertEquals(clobData, Bytes.toString(row1.get("CLOB_COL"), 0, clobData.length()));
     Assert.assertEquals(clobData, Bytes.toString(row2.get("CLOB_COL"), 0, clobData.length()));
+
+  }
+
+  @Test
+  @Category(SlowTests.class)
+  public void testBadNameBadConn() throws Exception {
+    String importQuery = "SELECT ID, NAME, SCORE, GRADUATED, TINY, SMALL, BIG, FLOAT_COL, REAL_COL, NUMERIC_COL, " +
+      "DECIMAL_COL, BIT_COL, DATE_COL, TIME_COL, TIMESTAMP_COL, BINARY_COL, BLOB_COL, CLOB_COL FROM my_table " +
+      "WHERE ID < 3";
+    String countQuery = "SELECT COUNT(ID) from my_table WHERE id < 3";
+
+    ETLStage sink = new ETLStage("Table", ImmutableMap.of(
+      "name", "outputTable",
+      Properties.Table.PROPERTY_SCHEMA, schema.toString(),
+      Properties.Table.PROPERTY_SCHEMA_ROW_FIELD, "ID"));
+
+    boolean badConn = false;
+
+    try {
+      ETLStage sourceBadConn = new ETLStage("Database", ImmutableMap.<String, String>builder()
+        .put(Properties.DB.CONNECTION_STRING, String.format("jdbc:hsqldb:hsql://localhost/%sWRONG",
+                                                            hsqlDBServer.database))
+        .put(Properties.DB.IMPORT_QUERY, importQuery)
+        .put(Properties.DB.COUNT_QUERY, countQuery)
+        .put(Properties.DB.JDBC_PLUGIN_NAME, "hypersql")
+        .build()
+      );
+
+      ETLBatchConfig etlConfigBadConn = new ETLBatchConfig("* * * * *", sourceBadConn, sink,
+                                                           Lists.<ETLStage>newArrayList());
+      AdapterConfig adapterConfigBadConn = new AdapterConfig("", TEMPLATE_ID.getId(),
+                                                             GSON.toJsonTree(etlConfigBadConn));
+      Id.Adapter adapterIdBadConn = Id.Adapter.from(NAMESPACE, "dbSourceTest");
+      AdapterManager managerBadConn = createAdapter(adapterIdBadConn, adapterConfigBadConn);
+
+      managerBadConn.start();
+      managerBadConn.waitForOneRunToFinish(5, TimeUnit.MINUTES);
+      managerBadConn.stop();
+    } catch (Exception e) {
+      badConn = true;
+    }
+    Assert.assertTrue(badConn);
```

Comment:
Move this to the end of the try block as `Assert.fail();` with a no-op in the catch block and get rid of the `badConn` boolean.